### PR TITLE
Reader Recommendations: delete inactive code

### PR DESCRIPTION
**Pull request by samouri**
I don't think this code has been actively accessible within wp-calypso for over a year.
It was a controller for a Recommendations page that used to exist that I think was mainly used for testing cold-start.

Lets get rid of it!

### CODE REVIEW

**Comment by bluefuton** [comment]
Thanks for cleaning this up @samouri!

I think we also need to remove it from `client/wordpress-com.js`. I get an error on build:

`ERROR in ./client/sections.js
Module not found: Error: Can't resolve 'reader/recommendations' in '/Users/chris/Sites/wp-calypso/client'
resolve 'reader/recommendations' in '/Users/chris/Sites/wp-calypso/client'
`

File: client/reader/recommendations/index.js [hunk 1]
```diff
@@ -1,44 +0,0 @@
...
-import { recommendedPosts } from './controller';
-import { preloadReaderBundle, sidebar, updateLastRoute } from 'reader/controller';
-import config from 'config';
-
-export default function() {
-	// Cold Start no longer exists - redirect to /
-	page( '/recommendations/start', '/' );
-
-	// Blog Recommendations no longer exists as its own page - redirect to /
-	page( '/recommendations', '/read/search' );
```
Comment (bluefuton) [inline client/reader/recommendations/index.js]: It'd be nice to keep this redirect. Can we add it to `legacyRedirects` in the main Reader controller perhaps?

Comment (samouri) [reply client/reader/recommendations/index.js]: Do we still link to there from anywhere?
I figured it made sense as a grace period, but I don't see a point to keeping around legacy redirects forever within client code

**Comment by samouri** [comment]
> I think we also need to remove it from client/wordpress-com.js. I get an error on build:

Thanks for spotting that for me!

File: client/reader/recommendations/index.js (hunk 1 above)
Comment (blowery) [reply client/reader/recommendations/index.js]: I'd still leave it for now.

**Pull request merged**